### PR TITLE
Post previews

### PR DESCRIPTION
**Pull request by josiahsprague**
Enable post previews to pass through to Solidus

### CODE REVIEW

**Comment by josiahsprague** [comment]
This should also add a `preview=true` param to the redirect along with the `_wp_json_nonce`, but those should only be added if it is a preview URL, normal URLs should be redirected without those parameters.

File: functions.php [hunk 1]
```diff
@@ -3,13 +3,25 @@
...
-  $url  = $_SERVER["SERVER_NAME"];
-  $url .= $_SERVER["REQUEST_URI"];
-
-  $redirect = preg_replace("/^(.*?)\.(.*)$/","$2", $url);
-
-  return 'http://'.$redirect;
+function getRedirect($postID) {
+  $domain = preg_replace('/^(.*?)\.(.*)$/','$2', $_SERVER['SERVER_NAME']);
+  if (!$postID) {
+    return 'http://'.$domain.$_SERVER["REQUEST_URI"]; // If we don't know which post, just preserve the path
```
Comment (pushred) [inline functions.php]: You could flatten this if/else since you're returning early.

**Comment by josiahsprague** [comment]
@pushred I'm going to merge this to master. It's tested, but I'm wondering your thoughts on merging in this redirects feature since it's kind of Solidus-specific and this nadando repo is public. I don't think anyone else is using it, but since this is becoming specific for us, should we change how we're handling this? Also, as a side-note, after merging this, I'm planning to tag v1.0.0.

File: functions.php [hunk 2]
```diff
@@ -3,13 +3,21 @@
...
 
-function getRedirect() {
-  $url  = $_SERVER["SERVER_NAME"];
-  $url .= $_SERVER["REQUEST_URI"];
-
-  $redirect = preg_replace("/^(.*?)\.(.*)$/","$2", $url);
-
-  return 'http://'.$redirect;
+function getRedirect($postID) {
+  $domain = preg_match('/bloodwater/',$_SERVER['SERVER_NAME'])?'bloodwater.org':preg_replace('/^(.*?)\.(.*)$/','$2', $_SERVER['SERVER_NAME']);
```
Comment (pushred) [inline functions.php]: What's the hardcoding for here?

**Comment by pushred** [comment]
> since this is becoming specific for us, should we change how we're handling this 

Maybe this should renamed and moved to solidusjs/wp-solidus?

**Comment by josiahsprague** [comment]
@pushred Are you still in favor or renaming this and moving it to solidusjs/wp-solidus? How about making it a private repo?

**Comment by pushred** [comment]
Yeah, a public repo is fine though

**Pull request merged**